nixos/modules/services/x11/desktop-managers/gnome3.xml

```diff
@@ -0,0 +1,250 @@
+  </para>
+ </section>
+
+ <section xml:id="sec-gnome-shell-extensions">
+  <title>Shell Extensions</title>
+
+  <para>
+   All shell extensions are packaged under the <literal>gnomeExtensions</literal> attribute.
+   Some packages that include shell extensions, like <literal>gnome3.gpaste</literal>, don't
+   have their extension decoupled under this attribute.
+  </para>
+
+  <para>
+   You can install them like any other package:
+  </para>
+
+<programlisting>
+<xref linkend="opt-environment.systemPackages"/> = [
+  gnomeExtensions.dash-to-dock
+  gnomeExtensions.gsconnect
+  gnomeExtensions.mpris-indicator-button
+];
+</programlisting>
+
+  <para>
+   Unfortunately, we lack a way for these to be managed in a completely declarative way.
+   So you have to enable them manually with an Extensions application.
+   It’s possible to use a GSettings Override for this on <literal>org.gnome.shell.enabled-extensions</literal>,
+   but those will only influence the default value.
+  </para>
+ </section>
+
+ <section xml:id="sec-gnome-gsettings-overrides">
+  <title>GSettings Overrides</title>
+
+  <para>
+   GSettings Overrides can be used to adjust the default settings for the GNOME desktop and apps.
+  </para>
+
+  <para>
+   Using this will allow you to pre-seed user defaults before you even start the session.
+  </para>
+
+  <para>
+   You can do this by using the <xref linkend="opt-services.xserver.desktopManager.gnome3.extraGSettingsOverrides"/> option.
+  </para>
+
+  <para>
+   Take note that whatever packages you want to override gsettings for, you need to add them to
+   <xref linkend="opt-services.xserver.desktopManager.gnome3.extraGSettingsOverridePackages"/>.
+  </para>
+
+  <para>
+   If you need to a tool to explore which gsettings you can set, you can use <literal>dconf-editor</literal>.
+  </para>
+
+  <section xml:id="sec-gnome-gsettings-overrides-example">
+   <title>Example</title>
+
+<programlisting>
+services.xserver.desktopManager.gnome3 = {
+  <xref linkend="opt-services.xserver.desktopManager.gnome3.extraGSettingsOverrides"/> = ''
+    # Change default background
+    [org.gnome.desktop.background]
+    picture-uri='file://${pkgs.nixos-artwork.wallpapers.mosaic-blue}'
+
+    # Favorite apps in gnome-shell
+    [org.gnome.shell]
+    favorite-apps=['org.gnome.Photos.desktop', 'org.gnome.Nautilus.desktop']
+  '';
+
+  <xref linkend="opt-services.xserver.desktopManager.gnome3.extraGSettingsOverridePackages"/> = [
+    pkgs.gsettings-desktop-schemas # for org.gnome.desktop
+    pkgs.gnome3.gnome-shell # for org.gnome.shell
+  ];
+};
+</programlisting>
+  </section>
+ </section>
+
+ <section xml:id="sec-gnome-faq">
+  <title>FAQ</title>
+
+  <section xml:id="sec-gnome-faq-can-i-use-lightdm-with-gnome">
+   <title>Can I use LightDM with GNOME?</title>
+
+   <para>
+    Yes you can, and any other display-manager in NixOS.
+   </para>
+
+   <para>
+    However, it doesn’t work correctly for the wayland session of GNOME yet, and
```

Comment:
IIRC it works with sddm at least, not sure what's the status for LightDM after the wayland session patches. Though I don't mind pushing people towards gdm (would be nice having `gnome3.enable = true` enable gdm by default, but that requires some reworking how we enable display-managers, so).